libs/base/Data/Vect.idr

```diff
@@ -782,6 +782,25 @@ range : {len : Nat} -> Vect len (Fin len)
 range {len=Z}   = []
 range {len=S _} = FZ :: map FS range
 
+--------------------------------------------------------------------------------
+-- Matrix transposition
+--------------------------------------------------------------------------------
+||| Specialises zipWith (::) to preserve linearity
+public export
+zipWithConsLinear : (1 _ : Vect a elem)
```

Comment:
Why not `lzipWith : (1 a -> 1 b -> c) -> 1 Vec n a -> 1 Vec n b -> Vec n c`?

Reply:
Since the `transpose` is `public export` I tried to keep what was there unchanged. Happy to go with the non-specialised version.

The more important question is whether `transpose` should really be `public export` if `transpose'` is `public export` and `transposeSpec` is `export`.

Reply:
I think we should opt for the more principled solution if we are going to
export the function. It may break things right now but we won't have to live
with the ad-hoc functions for the rest of our lives.

Reply:
What is the more principled solution?

Reply:
The generic `lzipWith` rather than a specialised version for `(::)`.

Reply:
Ah, I see. Done.

Reply:
Thanks!